src/sentry/static/sentry/app/components/dropdownAutoComplete.jsx

```diff
@@ -0,0 +1,165 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+import _ from 'lodash';
+import AutoComplete from './autoComplete';
+import Input from '../views/settings/components/forms/controls/input';
+
+class DropdownAutoComplete extends React.Component {
+  static propTypes = {
+    items: PropTypes.array,
+    isOpen: PropTypes.bool,
+    onSelect: PropTypes.func,
+    children: PropTypes.func,
+  };
+
+  static defaultProps = {
+    isOpen: false,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isOpen: this.props.isOpen,
+      selectedItem: undefined,
+    };
+  }
+
+  toggleMenu = () => this.setState({isOpen: !this.state.isOpen});
+
+  openMenu = e => this.setState({isOpen: true});
+
+  onSelect = selectedItem => {
+    this.setState({selectedItem});
+    if (this.props.onSelect) this.props.onSelect(selectedItem);
+    this.toggleMenu();
+  };
+
+  filterItems = (items, inputValue) =>
+    items.filter(item => {
+      return item.value.toLowerCase().indexOf(inputValue.toLowerCase()) > -1;
```

Comment:
still worried about this in a case where you want the result to be something really clear cut like an ID, but you want to display and search on something mushier like names and ids

for the search in our settings (not quite the same thing), we just throw the kitchen sink in there:
https://github.com/getsentry/sentry/blob/feat/actors/teamassign/src/sentry/static/sentry/app/utils/addForm.jsx#L6-L13

the label being jsx or html makes this tricker, so i don't necessarily have a solution and maybe we should just wait and see if it actually comes up. 
but it would be nice to have a one to one correspondence between the visible text and the autocomplete filtering

Reply:
the component returns the entire object you put in, so seems like you could just add whatever unique IDs you want and put the fuzzy stuff in the value. By limiting the search to the value we give the user of the component more control over what is searchable and what isn't, which feels like more of a win here since the input data can vary more.

fuzzy searching through something like a list of user badges seems very hard lol. I guess we could run an xml parser on it then flatten and join into a string 🤷‍♀️

Reply:
>the component returns the entire object you put in, so seems like you could just add whatever unique IDs you want and put the fuzzy stuff in the value. 
that sounds like a good solution to me

Reply:
i was also thinking something more constrained where you provide a stateless badge component instead of arbitrary jsx or html